model/test_results.go

```diff
@@ -216,31 +216,34 @@ func (t *TestResults) Download(ctx context.Context) ([]TestResult, error) {
 		false,
 	)
 	if err != nil {
-		return []TestResult{}, errors.Wrap(err, "problem creating bucket")
+		return nil, errors.Wrap(err, "problem creating bucket")
 	}
 
 	iter, err := bucket.List(ctx, "")
 	if err != nil {
-		return []TestResult{}, errors.Wrap(err, "problem listing bucket items")
+		return nil, errors.Wrap(err, "problem listing bucket items")
 	}
 
 	results := []TestResult{}
 	for iter.Next(ctx) {
 		r, err := iter.Item().Get(ctx)
 		if err != nil {
-			return []TestResult{}, errors.Wrap(err, "problem getting test result")
+			return nil, errors.Wrap(err, "problem getting test result")
 		}
 		defer func() {
-			_ = r.Close()
+			err = r.Close()
+			grip.Error(message.WrapError(err, message.Fields{
+				"message": "problem closing test results bucket iterator",
+			}))
```

Comment:
there's no need to change because I think this is all going to change but, this is really not all that different from: 
```
grip.Error(message.WrapError(r.Close(), "problem closing test results bucket iterator"))
```
Thought there's other things you can put in the message, for more context, including what test it's coming from/bucket name, etc.